### PR TITLE
Refactor crates

### PR DESCRIPTION
**Pull request by edgarriba**
- this is a **BREAKING COMMIT** towards 0.2
- converted package to `workspace`
- main crates now are `kornia-core` / `kornia-image` / `kornia-io` / `kornia-imgproc` / `kornia`
- the main namespace `kornia_rs` is now just `kornia`
- moved the main crates into `crates/*`
- examples have been factored to their own crates
- benches have been moved to their own subcrates

### CODE REVIEW

File: examples/normalize.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
-use kornia_rs::image::Image;
-use kornia_rs::io::functional as F;
+use kornia_image::Image;
+use kornia_io::functional as F;
```
Comment (johnnv1) [inline examples/normalize.rs:2]: why not keep using the top lib in the examples?

File: kornia-serve/Cargo.toml [hunk 2]
```diff
@@ -6,7 +6,7 @@ edition = "2021"
 [dependencies]
 axum = "0.7.5"
 env_logger = "0.11.3"
-kornia-rs = "0.1.3"
+kornia-rs = { workspace = true }
```
Comment (johnnv1) [inline kornia-serve/Cargo.toml:9]: when shipping it, will it be the right version this way?

File: benches/bench_color.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 use criterion::{black_box, criterion_group, criterion_main, BenchmarkId, Criterion};
```
Comment (johnnv1) [inline benches/bench_color.rs:1]: I would separate the benchmarks into subdirectories for each crate hre

File: .gitignore [hunk 4]
```diff
@@ -3,3 +3,4 @@
 target/
 docker/
 examples/*/Cargo.lock
+Cargo.lock
```
Comment (johnnv1) [inline .gitignore:6]: it's here, but the file is still on git

File: examples/normalize.rs (hunk 1 above)
Comment (edgarriba) [reply examples/normalize.rs:2]: still figuring out how to recover the top level api -- maybe it's gonna be a big breaking change. Not clear yet

Comment (edgarriba) [reply examples/normalize.rs]: Fixed

File: README.md [hunk 5]
```diff
@@ -123,7 +123,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let gray_resize: Image<f32, 1> = kornia_rs::resize::resize_native(
         &gray,
-        kornia_rs::image::ImageSize {
+        kornia_image::ImageSize {
```
Comment (edgarriba) [inline README.md:126]: Check imports in docs

File: crates/kornia-imgproc/src/color/gray.rs [hunk 6]
```diff
@@ -33,7 +33,7 @@ const BW: f64 = 0.114;
 ///     vec![0f32; 4 * 5 * 3],
 /// )
 /// .unwrap();
-/// let gray: Image<f32, 1> = kornia_rs::color::gray_from_rgb(&image).unwrap();
+/// let gray: Image<f32, 1> = kornia_imgproc::color::gray_from_rgb(&image).unwrap();
```
Comment (edgarriba) [inline crates/kornia-imgproc/src/color/gray.rs:36]: Check Import

**Comment by mweinelt** [comment]
Was the removal of Cargo.lock intentional? For nixpkgs this creates a problem, since we now need to provide our own to be able to reproducibly fetch the same dependencies.

**Comment by edgarriba** [comment]
We can add again

**Comment by mweinelt** [comment]
That would be appreciated.

**Pull request merged**